Mage.Sets/src/mage/cards/m/MirriWeatherlightDuelist.java

```diff
@@ -78,12 +78,13 @@ public boolean applies(Permanent permanent, Ability source, Game game) {
     }
 
     @Override
-    public boolean canBlock(Permanent attacker, Permanent blocker, Ability source, Game game, boolean canUseChooseDialogs) {
+    public boolean canBlock(Permanent attacker, Permanent newBlocker, Ability source, Game game, boolean canUseChooseDialogs) {
         if (attacker == null) {
             return true;
         }
-        for (UUID creature : game.getCombat().getBlockers()) {
-            if (game.getPlayer(game.getPermanent(creature).getOwnerId()).hasOpponent(attacker.getControllerId(), game)) {
+        for (UUID creatureId : game.getCombat().getBlockers()) {
+            Permanent existingBlocker = game.getPermanent(creatureId);
+            if (game.getPlayer(existingBlocker.getControllerId()).hasOpponent(attacker.getControllerId(), game) && existingBlocker.isControlledBy(newBlocker.getControllerId())) {
```

Comment:
So the first check is to see if the existingBlocker is controller by one of your opponents,
and the second is to check if newBlocker and the existingBlocker have the same controller?

Reply:
I don't know, I just cargo-culted the first check from the existing implementation (with a fix to the logic.) Theoretically if two-headed giant was supported, you could have a limited range of influence two-headed giant game where an opponent who is outside of your range is being attacked by your teammate, and that opponent's creatures shouldn't be effected by Mirri's ability?